### PR TITLE
feat(nextjs): Deprecate `clerkClient` singleton into a proxy for dynamic keys 

### PR DESCRIPTION
**Pull request by LauraBeatris**
## Description

Part of SDK-1810

Refactors `clerkClient` to be a proxy function from `createClerkClient` to read dynamic keys from the request. 

`clerkClient` as an object will get deprecated and removed in the next major version. 

## Checklist

- [X] `npm test` runs as expected.
- [X] `npm run build` runs as expected.
- [X] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [X] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

**Comment by LauraBeatris** [comment]
I'll also make sure to update the docs to mention `clerkClient()` instead of `clerkClient` https://clerk.com/docs/references/nextjs/get-auth#usage-with-clerk-client

**Comment by LauraBeatris** [comment]
This is currently not working due to the `Object.assign` overriding pretty much all properties from the singleton. I'll continue to investigate a solution tomorrow.

CleanShot 2024-06-19 at 16 13 50@2x

File: packages/nextjs/src/server/clerkClient.ts [hunk 1]
```diff
@@ -1,5 +1,9 @@
+import type { ClerkClient } from '@clerk/backend';
 import { createClerkClient } from '@clerk/backend';
+import { constants } from '@clerk/backend/internal';
+import { deprecated } from '@clerk/shared';
```
Comment (panteliselef) [inline packages/nextjs/src/server/clerkClient.ts:4]: Nitpicking
```suggestion
import { deprecated } from '@clerk/shared/deprecated';
```

**Comment by LauraBeatris** [comment]
I've ended up doing a rebase against `main` that messed up this whole branch... I'll close this one and open another which is going to be easier than solving here 😢

**Pull request closed without merging**